### ISSUE TITLE
v3.x: Unclear error when no rules found

### ISSUE THREAD

**Issue by shinnn** [stylelint/stylelint#584]
With stylelint v2.x, it shows a clear error message `No rules found within configuration` when the configuration doesn't include any rules.

```js
// stylelint v2.3.7
'use strict';
const stylelint = require('stylelint');

stylelint.lint({code: 'a{}'}).catch(err => {
  console.error(err.stack);
});
```

```
Error: No rules found within configuration
    at exports.default (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/stylelint/dist/utils/configurationError.js:8:13)
    at /Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/stylelint/dist/postcssPlugin.js:52:43
    at LazyResult.run (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/postcss/lib/lazy-result.js:206:20)
    at /Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/postcss/lib/lazy-result.js:120:37
    at LazyResult.asyncTick (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/postcss/lib/lazy-result.js:134:15)
    at processing.Promise.then._this2.processed (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/postcss/lib/lazy-result.js:160:20)
    at LazyResult.async (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/postcss/lib/lazy-result.js:157:27)
    at LazyResult.then (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/postcss/lib/lazy-result.js:79:21)
    at lint (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/stylelint/dist/standalone.js:103:186)
    at pop (/Users/Shinnosuke/github/MyRepositories/stylelint-vscode/node_modules/queue-async/queue.js:24:14)
```

With v3.x, it shows less clear error by which we can't understand what should the user do to fix the configuration error.

```js
// stylelint v3.1.3
'use strict';
const stylelint = require('stylelint');

stylelint.lint({code: 'a{}'}).catch(err => {
  console.error(err.stack);
});
```

```
TypeError: Cannot read property 'filepath' of null
    at /Users/me/test/node_modules/stylelint/dist/buildConfig.js:32:50
```

**Comment by davidtheclark** [comment]
The same error message `No rules found within configuration` should show up at the same time inv3. I think you're actually suffering from a different bug, related to #582. I am pushing a fix very soon.

**Comment by davidtheclark** [comment]
@shinnn : Please try v3.1.4 and see if your problem is solved.

**Comment by shinnn** [comment]
@davidtheclark Confirmed it's fixed in v3.1.4. Thanks!
